src/main/java/org/kairosdb/datastore/cassandra/cache/AbstractByteBufferCache.java

```diff
@@ -0,0 +1,43 @@
+package org.kairosdb.datastore.cassandra.cache;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.google.common.annotations.VisibleForTesting;
+import org.kairosdb.datastore.cassandra.cache.persistence.GeneralHashCacheStore;
+
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.hash.Hashing.murmur3_128;
+import static net.openhft.hashing.LongHashFunction.xx;
+
+public abstract class AbstractByteBufferCache {
+    private static final int MURMUR_SEED = 0xDEADBEEF;
+    private static final int XX_SEED = 0xCAFEBABE;
+    private static final BigInteger MASK = BigInteger.valueOf(Long.MAX_VALUE);
+
+    protected final LoadingCache<BigInteger, Object> outerLayerCache;
+
+    protected AbstractByteBufferCache(final GeneralHashCacheStore cacheStore,
+                                final int maxSize, final int ttlInSeconds) {
+        this.outerLayerCache = Caffeine.newBuilder()
+                .initialCapacity(maxSize / 3 + 1)
+                .maximumSize(maxSize)
+                .expireAfterWrite(ttlInSeconds, TimeUnit.HOURS)
```

Comment:
`ttlInSeconds` holds number of seconds, right? And time unit is `TimeUnit.HOURS`. What if someones decides to pass `100500` as a TTL?

Reply:
Good catch